hare/hare.go

```diff
@@ -172,13 +172,18 @@ func (h *Hare) onTick(id types.LayerID) {
 		set.Add(Value{b})
 	}
 
-	instid := InstanceId(id)
-
-	cp := h.factory(h.config, instid, set, h.rolacle, h.sign, h.network, h.outputChan)
-	cp.SetInbox(h.broker.Register(cp.Id()))
+	instId := InstanceId(id)
+	c, err := h.broker.Register(instId)
+	if err != nil {
+		h.Warning("Could not register CP to broker err=%v", err)
+		return
+	}
+	cp := h.factory(h.config, instId, set, h.rolacle, h.sign, h.network, h.outputChan)
+	cp.SetInbox(c)
 	e := cp.Start()
 	if e != nil {
 		h.Error("Could not start consensus process %v", e.Error())
+		return
```

Comment:
you need to unregister in such case, this is a memory leak

Reply:
Very true. Fixed.